### ISSUE TITLE
Enhance examples/ directory with comprehensive visualization and processing notebooks

### ISSUE THREAD

**Issue by tallamjr** [tallamjr/evlib#11]
# Enhance examples/ directory with comprehensive visualization and processing notebooks

## Overview

The current `examples/` directory contains basic usage scripts and two visualization notebooks, but lacks comprehensive, modern examples that showcase evlib's full capabilities for event camera processing and visualization. This issue proposes creating enhanced examples that demonstrate practical workflows and advanced techniques.

### Existing Examples (Good Foundation)
- **Notebooks**: `events_viz_0.ipynb`, `events_viz_1.ipynb` - Basic visualization exercises
- **Scripts**: Simple usage examples (`simple_example.py`, `reader_showcase.py`, etc.)
- **Coverage**: Basic representations, data loading, simple processing

### Current Gaps
1. **Limited evlib integration**: Notebooks mostly use manual implementations instead of evlib functions
2. **Outdated visualization techniques**: Basic matplotlib without modern best practices
3. **Missing practical workflows**: No end-to-end processing pipelines
4. **No performance comparison**: evlib vs traditional methods not demonstrated
5. **Limited real-world applications**: Focus on academic exercises rather than practical use cases

### 1. **Comprehensive Visualization Notebook** 
**File**: `examples/comprehensive_visualization.ipynb`

**Content**:
- Modern evlib API usage for all representations
- Interactive visualizations with plotly/bokeh
- Real-time processing examples
- Performance comparisons (evlib vs manual implementations)
- Multiple dataset formats (EVT2, HDF5, text)
- Advanced visualizations:
  - Time surfaces with proper temporal decay
  - 3D spatiotemporal point clouds with interactive controls
  - Event stream analysis and statistics
  - Motion flow visualization
  - Polarity-based edge detection

### 2. **Processing Pipeline Notebook**
**File**: `examples/processing_pipeline_showcase.ipynb`

**Content**:
- Complete data loading → processing → visualization pipeline
- Using evlib representations (stacked histograms, voxel grids, smooth representations)
- Event filtering and preprocessing
- Integration with PyTorch/neural networks
- Performance optimization techniques
- Memory-efficient processing for large datasets
- Real-world applications (object detection, tracking, SLAM)

### 3. **Format Support Showcase**
**File**: `examples/format_support_showcase.ipynb`

**Content**:
- Comprehensive format reader demonstration
- Automatic format detection examples
- Performance comparison across formats
- Handling different polarity encodings (0/1 vs -1/1)
- Large file processing strategies
- Data conversion utilities

### 4. **Advanced Techniques Notebook**
**File**: `examples/advanced_techniques.ipynb`

**Content**:
- Event stream augmentation and transformations
- Temporal filtering and noise reduction
- Multi-resolution processing
- Event-based optical flow
- Integration with traditional computer vision
- Custom representation development

### 5. **Performance Benchmarking**
**File**: `examples/performance_benchmarks.ipynb`

**Content**:
- evlib vs traditional numpy/pandas performance
- Memory usage optimization
- Processing speed comparisons
- Best practices for large datasets
- Polars integration benefits demonstration

### 6. **Real-world Applications**
**File**: `examples/applications_showcase.ipynb`

**Content**:
- Object tracking with event cameras
- High-speed motion analysis
- Low-light event processing
- Stereo event camera processing
- Neuromorphic computing integration

### Dependencies
```python
# Core
evlib
numpy >= 1.24.0
polars >= 1.30.0

# Visualization
matplotlib >= 3.7.0
plotly >= 5.0.0
bokeh >= 3.0.0 (optional)
seaborn >= 0.12.0

# Processing
torch >= 2.0.0 (optional)
opencv-python >= 4.8.0 (optional)

# Interactive
jupyter
ipywidgets
```

### Notebook Structure
Each notebook should include:
1. **Clear objectives** - What you'll learn
2. **Prerequisites** - Required knowledge/data
3. **Theory section** - Brief background
4. **Hands-on code** - Well-commented implementations
5. **Performance analysis** - Speed/memory comparisons
6. **Exercises** - Interactive learning opportunities
7. **References** - Papers, documentation links

### Phase 1: Foundation (Priority: High)
1. Create `comprehensive_visualization.ipynb` with modern evlib usage
2. Update existing notebooks to use evlib functions instead of manual implementations
3. Add interactive visualizations and performance comparisons

### Phase 2: Advanced Content (Priority: Medium)
1. Develop `processing_pipeline_showcase.ipynb`
2. Create `format_support_showcase.ipynb`
3. Add performance benchmarking examples

### Phase 3: Applications (Priority: Low)
1. Develop `advanced_techniques.ipynb`
2. Create `applications_showcase.ipynb`
3. Add specialized domain examples

## Success Metrics

1. **User engagement**: Notebook execution without errors on different systems
2. **Educational value**: Clear progression from basic to advanced concepts
3. **Performance demonstration**: Visible evlib advantages over manual methods
4. **Practical utility**: Real-world applicable examples
5. **Documentation quality**: Self-contained, well-explained examples

### Data Requirements
- Use existing datasets in `data/` directory
- Provide fallback synthetic data generation
- Include data download utilities for missing datasets
- Support both small (tutorial) and large (realistic) datasets

### Visualization Best Practices
- Responsive, interactive plots where beneficial
- Clear colormaps and accessibility considerations  
- Export capabilities for figures
- Multiple visualization backends support

### Code Quality
- Follow project coding standards (Black formatting)
- Include type hints and docstrings
- Error handling and fallback modes
- Cross-platform compatibility

### Testing
- Automated notebook testing with `pytest --nbmake`
- Performance regression testing
- Multiple Python version compatibility
- Memory usage validation

## Related Issues/PRs
- Links to relevant evlib feature requests
- Visualization improvements
- Performance optimization work

---

**Labels**: `enhancement`, `documentation`, `examples`, `visualization`, `good-first-issue`
**Assignees**: Documentation team, visualization experts
**Milestone**: Next release

This enhancement will significantly improve evlib's accessibility and demonstrate its powerful capabilities to new users while providing practical resources for advanced users.
